tests/template_tests/test_context.py

```diff
@@ -30,6 +30,15 @@ def test_context(self):
             self.assertEqual(c['a'], 3)
         self.assertEqual(c['a'], 1)
 
+        with c.update({}):
```

Comment:
I think it would be better to make this a separate test method as the test is getting a bit long and unfocused (feel free to add a method for the push() context manager tests as well?)